### DISCUSSION TITLE
Get next value from sequence in database

### DISCUSSION THREAD

**Discussion by zbouchlih** [serenity-is/Serenity#6067] [Help] [answered]
Hey All,

I would like to use this query "SELECT NEXT VALUE FOR SequenceIndex"  (sequenceIndex is a sequence in my database) to get a sequence number that I will use to fill in a field automatically in my form.

1- I suppose I can't do that with an SqlQuery Object in Serenity,  there a way to call this query ?

2- And I should do that inside a request handler right ? that will call a helper with the query ?

Thank you so much for your help

**Comment by wezmag** [comment, marked as answer]
Serenity uses Dapper for database access, therefore you can also use Dapper to call native SQL query.

This is how I did in the repository class in my previous (.NET Framework) project:
``` csharp
using Dapper;
using System.Linq;

protected override void BeforeSave()
{
    if (IsCreate)
    { 
        var newId = this.Connection.Query<int>("SELECT NEXT VALUE FOR SequenceIndex").Single();
        this.Request.Entity.SomeIdField = newId;
    }
    base.BeforeSave();
}
```

I think in the new (.net core) project, you should put above code in the SaveHandler.

**Reply by zbouchlih** [reply]
Thanks a lot :)
